R/pkg/R/DataFrame.R

```diff
@@ -1136,9 +1136,17 @@ setMethod("collect",
 
                   # Note that "binary" columns behave like complex types.
                   if (!is.null(PRIMITIVE_TYPES[[colType]]) && colType != "binary") {
-                    vec <- do.call(c, col)
+                    valueIndex <- which(!is.na(col))
+                    if (length(valueIndex) > 0 && valueIndex[1] > 1) {
+                      colTail <- col[-(1 : (valueIndex[1] - 1))]
+                      vec <- do.call(c, colTail)
+                      classVal <- class(vec)
+                      vec <- c(rep(NA, valueIndex[1] - 1), vec)
+                      class(vec) <- classVal
```

Comment:
Hmm, what happened here?
if you want to drop the NA and use the rest to infer the class you can do `col[!is.na(col)]`